design/design-doc-stacks.md

```diff
@@ -195,42 +200,69 @@ spec:
+
+### Stack Files
+
 * `app.yaml`: This file is the general metadata and information about the Stack, such as its name, description, version, owners, etc.  This metadata will be saved in the `Extension` record's spec fields.
 * `install.yaml`: This file contains the information for how the custom controller for the Stack should be installed into Crossplane.  Initially, only simple `Deployment` based controllers will be supported, but eventually other types of implementations will be supported as well, e.g., templates, functions/hooks, templates, a new DSL, etc.
-* `resources` directory: This directory contains all the CRDs and optional metadata about them.  These CRDs are the types that the custom controller implements the logic for.  They will be directly installed into Crossplane so that users can create instances of them to start consuming their new Stack functionality.
-* `ui-schema.yaml`: UI metadata that will be transformed and annotated according to the [Stack UI Metadata One Pager](one-pager-stack-ui-metadata.md)
+* `icon.svg`: This file (or `icon.png`, `icon.jpg`, `icon.gif`, or potentially other supported filenames, TBD) will be used in a visual context when listing or describing this stack.  The preferred formats/filenames are `svg`, `png`, `jpg`, `gif` in that order (if multiple files exist).  For bitmap formats, the width to height ratio should be 1:1. Limitations may be placed on the acceptable file dimensions and byte size (TBD).
+* `resources` directory: This directory contains all the CRDs and optional metadata about them.
+  * `group.yaml`: Related Stack resources (CRDs) can be described at a high level within a group directory using this file.
+  * `resource.yaml` and `icon.svg`: Files that describe the resource with descriptions, titles, or images, may be used to inform out-of-cluster or in-cluster Stack managing tools.  This may affect the annotations of the `Extension` record or the Resource CRD (TBD).
+  * `ui-schema.yaml`: UI metadata that will be transformed and annotated according to the [Stack UI Metadata One Pager](one-pager-stack-ui-metadata.md)
+  * `*crd.yaml`: These CRDs are the types that the custom controller implements the logic for.  They will be directly installed into Crossplane so that users can create instances of them to start consuming their new Stack functionality.  Notice that the filenames can vary from `very.descriptive.name.crd.yaml` to `crd.yaml`.
+  Multiple CRDs can reside in the same file.  These CRDs may also be pre-annotated at build time with annotations describing the `resource.yaml`, `icon.svg`, and `ui-schema.yaml` files to avoid bundling additional files and incurring a minor processing penalty at runtime.
+
+The minimum required file tree for a single tool, such as the Stack Manager, could be condensed to the following:
+
+```text
+.registry/
+├── app.yaml
```

Comment:
these minimum package contents would result in no "implementation" to manage the CRD.  Do you think an `install.yaml` and probably `rbac.yaml` would be needed as well?

Reply:
These files were and still are documented as optional.  For a _minimal_ stack layout I left them out.  I considered adding some wording to accent that this is likely too minimal.

I'll revisit this since it caught your eye too.

Reply:
I would suggest that all of them are optional, including `app.yaml`. We could **recommend** that people include more than zero of them if they're creating something that would be following best practices, and want to make a high-quality stack that other people will be consuming, but I feel it is also a valid use-case to have a stack with no metadata and no install or rbac requirements. Making any of these files explicitly required seems like an artificial constraint to me.

I suppose using my argument, I'm not sure `crd.yaml` should be considered required either.

In general, I want to make sure we aren't unnecessarily conflating structural requirements with recommended patterns. For example, just because all of our documented examples use a certain structure doesn't mean we need to strictly enforce that the structure is followed.

Reply:
Revised in the latest squashed commit.